Sources/Hedera/MirrorNodeGateway.swift

```diff
@@ -0,0 +1,114 @@
+    }
+
+    internal static func forNetwork(_ mirrorNetwork: [String], _ ledgerId: LedgerId?) throws -> MirrorNodeGateway {
+        let mirrorNodeUrl = try MirrorNodeRouter.getMirrorNodeUrl(mirrorNetwork, ledgerId)
+
+        return .init(mirrorNodeUrl: mirrorNodeUrl)
+    }
+
+    internal func getAccountInfo(_ idOrAliasOrEvmAddress: String) async throws -> [String: Any] {
+        let fullApiUrl = MirrorNodeRouter.buildApiUrl(
+            self.mirrorNodeUrl, MirrorNodeRouter.accountsRoute, idOrAliasOrEvmAddress)
+
+        let responseBody = try await queryFromMirrorNode(fullApiUrl)
+
+        let jsonObject = try await deserializeJson(responseBody)
+
+        return jsonObject
+    }
+
+    internal func getContractInfo(_ idOrAliasOrEvmAddress: String) async throws -> [String: Any] {
+        let fullApiUrl = MirrorNodeRouter.buildApiUrl(
+            self.mirrorNodeUrl, MirrorNodeRouter.contractsRoute, idOrAliasOrEvmAddress)
+
+        let responseBody = try await queryFromMirrorNode(fullApiUrl)
+
+        let jsonObject = try await deserializeJson(responseBody)
+
+        return jsonObject
+    }
+
+    internal func getAccountTokens(_ idOrAliasOrEvmAddress: String) async throws -> [String: Any] {
+        let fullApiUrl = MirrorNodeRouter.buildApiUrl(
+            self.mirrorNodeUrl, MirrorNodeRouter.tokensAccountRoute, idOrAliasOrEvmAddress)
+
+        let responseBody = try await queryFromMirrorNode(fullApiUrl)
+
+        let jsonObject = try await deserializeJson(responseBody)
+
+        return jsonObject
+    }
+
+    private func queryFromMirrorNode(_ apiUrl: String) async throws -> String {
+        let httpClient = HTTPClient(eventLoopGroupProvider: .singleton)
+        defer {
+            try? httpClient.syncShutdown()
+        }
+
+        if apiUrl.contains("127.0.0.1:5551") {
+            try await Task.sleep(nanoseconds: 1_000_000_000 * 3)
+        }
+
+        let request = HTTPClientRequest(url: apiUrl)
+
+        let response: HTTPClientResponse = try await httpClient.execute(request, timeout: .seconds(30))
+
+        let body = try await response.body.collect(upTo: 1024 * 1024)
+        let bodyString = String(decoding: body.readableBytesView, as: UTF8.self)
+
+        return bodyString
+    }
+
+    func deserializeJson(_ responseBody: String) async throws -> [String: Any] {
+        guard let jsonData = responseBody.data(using: .utf8) else {
+            throw NSError(
+                domain: "InvalidResponseError", code: -1,
+                userInfo: [NSLocalizedDescriptionKey: "Response body is not valid UTF-8"])
+        }
+
+        guard let jsonObject = try JSONSerialization.jsonObject(with: jsonData, options: []) as? [String: Any] else {
+            throw NSError(
+                domain: "InvalidResponseError", code: -1,
+                userInfo: [NSLocalizedDescriptionKey: "Response body is not a valid JSON object"])
+        }
+
+        return jsonObject
+    }
```

Comment:
Look for `JSONDecoder` and `Codable`:
- https://www.hackingwithswift.com/read/7/3/parsing-json-using-the-codable-protocol
- https://developer.apple.com/documentation/foundation/jsondecoder
Also, this function doesn't need to be async.